pkgs/build-support/gradle/update-deps.nix

```diff
@@ -0,0 +1,152 @@
+{ lib
+, pkgs
+, runtimeShell
+, stdenv
+, srcOnly
+, writeTextFile
+, writeShellScript
+, path
+, bubblewrap
+, coreutils
+, curl
+, jq
+, mitm-cache
+, nix
+, openssl
+, procps
+, python3
+}:
+
+let
+  getPkg = attrPath:
+    lib.getAttrFromPath
+      (lib.splitString "." (toString attrPath))
+      pkgs;
+in
+lib.makeOverridable (
+# the derivation to update
+{ pkg ? getPkg attrPath
+, pname ? null
+, attrPath ? pname
+# bwrap flags (this will be put in bash as-is)
+# this is relevant for downstream users
+, bwrapFlags ? "--ro-bind \"$PWD\" \"$PWD\""
+# run before fetching the dependencies (in sandbox)
+, preBuild ? ""
+# the gradle commands for fetching the necessary dependencies
+, script ? "gradle nixDownloadDeps"
+# run after fetching the dependencies (in sandbox)
+, postBuild ? ""
```

Comment:
would it maybe be better to make `preBuild`, `script`, `postBuild` an actual phase like `gradleUpdatePhase` or `gradleUpdateDepsPhase` or `gradleFetchDepsPhase` or `gradleFetchPhase`?

It would allow overriding these commands, and `preBuild` would turn from a magical argument (it's distinct from the derivation's `preBuild`, which is called in the update script as well) into a proper `preGradleUpdate` hook, but it may be weird that the phase is never used when actually building the package.

Reply:
Maybe call it gradleUpdateCommands or Script or something to better indicate that it isn't run by the normal phases?